src/main/java/pong/Ball.java

```diff
@@ -1,4 +1,56 @@
 package pong;
 
 public class Ball {
+    private Paddle humanPaddle;
+    private Paddle compPaddle;
+
+    private double x;
+    private double y;
+    private double velX;
+    private double velY;
+    private final int MID_OF_BALL = PongView.BALL_WIDTH / 2;
+
+    public Ball(Paddle humanPaddle, Paddle compPaddle) {
+        //first position - middle of the board.
+        x = PongFrame.WIDTH / 2;
+        y = PongFrame.HEIGHT / 2;
+        velY = 1;
+        velX = -2;
+        this.humanPaddle = humanPaddle;
+        this.compPaddle = compPaddle;
+    }
+
+    public int getX() {
+        return (int) x;
+    }
+
+    public int getY() {
+        return (int) y;
+    }
+
+    public void move() {
+        x += velX;
+        y += velY;
+        checkPaddleCollide();
+        checkWallCollide();
+    }
+
+    public void checkWallCollide() {
+        if (this.y >= PongFrame.HEIGHT - MID_OF_BALL || this.y <= MID_OF_BALL){
+            velX = -velX;
+        }
```

Comment:
I thought this collision thing would be much more complicated, but I saw the idea to negate the x or y values in a few places online. [This guy](https://www.youtube.com/watch?v=xIqeK2hzx1I) did a great job walking me through how to get it to move or what to do on collision in his video. I hadn't thought of just incrementing the x and y values to get it to move ( as demoed in move() method), And I thought for the collisions I would need to figure out angles or something like that.